rotkehlchen/api/v1/resources.py

```diff
@@ -2311,26 +2313,27 @@ def post(self, file: FileStorage) -> Response:
         return response
 
 
-class DBSnapshotExportingResource(BaseMethodView):
-    post_schema = SnapshotExportingSchema()
-
-    @require_loggedin_user()
-    @use_kwargs(post_schema, location='json')
-    def post(self, timestamp: Timestamp, path: Path) -> Response:
-        return self.rest_api.export_user_db_snapshot(timestamp=timestamp, path=path)
-
-
-class DBSnapshotDownloadingResource(BaseMethodView):
-    post_schema = SnapshotTimestampQuerySchema()
+class DBSnapshotsResource(BaseMethodView):
+    get_schema = SnapshotQuerySchema()
+    upload_schema = SnapshotImportingSchema()
+    patch_schema = SnapshotEditingSchema()
+    delete_schema = SnapshotTimestampQuerySchema()
 
     @require_loggedin_user()
-    @use_kwargs(post_schema, location='json')
-    def post(self, timestamp: Timestamp) -> Response:
-        return self.rest_api.download_user_db_snapshot(timestamp=timestamp)
-
+    @use_kwargs(get_schema, location='json_and_query_and_view_args')
+    def get(
+        self,
+        action: Optional[str],
+        timestamp: Timestamp,
+        path: Optional[Path],
+    ) -> Response:
+        if action == 'export':
+            if path is not None:
```

Comment:
If this is due to mypy it should be an assertion. Or a #type:ignore 
comment explaining why it's there.

This is the wrong approach since it makes no sense to have this if check. As It should never happen. Its mere existence indicates that it's possible to happen. And then what? What happens at the else? `None` is returned.